src/App.tsx

```diff
@@ -1,59 +1,115 @@
-import { useEffect, useState } from 'react';
-
+import React, { useState } from 'react';
 import './App.scss';
+
 import usersFromServer from './api/users';
 import todosFromServer from './api/todos';
-import { TodoList } from './components/TodoList';
-import { Todo } from './types/Todo';
-import { TodoForm } from './components/TodoForm';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { User } from './types/User';
+import { Todos } from './types/Todos';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
 
-const todoWithUser: Todo[] = todosFromServer.map(todo => ({
-  ...todo,
-  user: usersFromServer.find(user => user.id === todo.userId) || null,
-}));
+  return foundUser || null;
+}
+
+function InitialArray(): Todos[] {
+  return todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
+}
 
 export const App = () => {
-  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sectionValue, setSectionValue] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [CopyToDoList, setCopyToDoList] = useState<Todos[]>(InitialArray());
```

Comment:
It's not a copy, you just set the initial value as the existing array
```suggestion
  const [todoList, setToDoList] = useState<Todos[]>(InitialArray());
```